src/Resources/ResourceManager/SdkModels/Deployments/PSWhatIfOperationResult.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels.Deployments
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Management.ResourceManager.Models;
+
+    public class PSWhatIfOperationResult
+    {
+        private readonly WhatIfOperationResult whatIfOperationResult;
+
+        public PSWhatIfOperationResult(WhatIfOperationResult whatIfOperationResult)
+        {
+            this.whatIfOperationResult = whatIfOperationResult;
+        }
+
+        public string Status => this.whatIfOperationResult.Status;
```

Comment:
>Status [](start = 59, length = 6)

I believe these are defined in the SDK namespace. Do we need to return something that's defined in the PowerShell namespace (Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels.Deployments)?

Reply:
Yes, but I think this is already under that namespace? `WhatIfOperationResult` is defined in the SDK namespace and `PSWhatIfOperationResult` is the wrapper type that we return to users.

Reply:
I see. I'm not sure about the best practice here. maybe just check what the other cmdlets do.

Reply:
Found https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#returning-wrapped-sdk-types. I think we are following the best practice :p